src/main/java/org/isf/menu/service/UserIoOperationRepository.java

```diff
@@ -37,15 +37,29 @@ public interface UserIoOperationRepository extends JpaRepository<User, String> {
 
 	List<User> findAllByOrderByUserNameAsc();
 
+	List<User> findAllByDeletedOrderByUserNameAsc(boolean deleted);
+
 	User findByUserName(String userName);
 
+	List<User> findAllByDeleted(boolean deleted);
+
+	User findByUserNameAndDeleted(String userName, boolean deleted);
+
 	@Query(value = "select user from User user where user.userGroupName.code=:groupId order by user.userName")
 	List<User> findAllWhereUserGroupNameByOrderUserNameAsc(@Param("groupId") String groupId);
 
+	@Query(value = "select user from User user where user.userGroupName.code=:groupId and deleted=:deleted order by user.userName")
+	List<User> findByDeletedWhereUserGroupNameByOrderUserNameAsc(@Param("groupId") String groupId, @Param("deleted") boolean deleted);
+
```

Comment:
These methods seems wrongly named:
- `findAll` --> when returning a list
- `By` --> standards is to use By (in place of Where)
- `FieldsNames` like parameters using the same order

```suggestion
	@Query(value = "select user from User user where user.userGroupName.code=:groupId order by user.userName")
	List<User> findAllByUserGroupIdOrderByUserNameAsc(@Param("groupId") String groupId);

	@Query(value = "select user from User user where user.userGroupName.code=:groupId and deleted=:deleted order by user.userName")
	List<User> findAllByUserGroupIdAndDeletedOrderByUserNameAsc(@Param("groupId") String groupId, @Param("deleted") boolean deleted);

```

Reply:
Nice catch !